### PR TITLE
Refactor SimpleOrderService#collectOrderItemDetails method to use Java Stream API

### PR DESCRIPTION
**Pull request by Andrii77711**
solve issue https://github.com/Andrii77711/order-management-service/issues/4

### CODE REVIEW

File: src/main/java/com/to/ordermanagementservice/service/impl/SimpleOrderService.java [hunk 1]
```diff
@@ -6,11 +6,14 @@
 import com.to.ordermanagementservice.entity.OrderItem;
 import com.to.ordermanagementservice.repository.OrderItemRepository;
 import com.to.ordermanagementservice.repository.OrderRepository;
+import com.to.ordermanagementservice.repository.impl.InMemoryOrderRepository;
```
Comment (taras-oglabyak) [inline src/main/java/com/to/ordermanagementservice/service/impl/SimpleOrderService.java]: is this import needed?

your IDE has the functionality to organize imports and format code, please check the manual to understand how it works

File: src/main/java/com/to/ordermanagementservice/service/impl/SimpleOrderService.java [hunk 2]
```diff
@@ -6,11 +6,14 @@
...
 import com.to.ordermanagementservice.service.OrderService;
 import org.springframework.stereotype.Service;
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Set;
```
Comment (taras-oglabyak) [inline src/main/java/com/to/ordermanagementservice/service/impl/SimpleOrderService.java]: unused import

File: src/main/java/com/to/ordermanagementservice/service/impl/SimpleOrderService.java [hunk 3]
```diff
@@ -46,20 +49,22 @@ private OrderDetails collectOrderDetails(Order order) {
...
+        orderItemDetails.setUpdateAt(orderItem.getUpdatedAt());
+        orderItemDetails.setCreatedAt(orderItem.getCreatedAt());
+        orderItemDetails.setPrice(orderItem.getPrice());
+        orderItemDetails.setQuantity(orderItem.getQuantity());
+        orderItemDetails.setProductId(orderItem.getProductId());
+        return orderItemDetails;
     }
 
+
+
```
Comment (taras-oglabyak) [inline src/main/java/com/to/ordermanagementservice/service/impl/SimpleOrderService.java]: do we need these extra lines?

File: src/main/java/com/to/ordermanagementservice/service/impl/SimpleOrderService.java [hunk 4]
```diff
@@ -6,13 +6,11 @@
...
 import com.to.ordermanagementservice.repository.OrderItemRepository;
 import com.to.ordermanagementservice.repository.OrderRepository;
-import com.to.ordermanagementservice.repository.impl.InMemoryOrderRepository;
 import com.to.ordermanagementservice.service.OrderService;
 import org.springframework.stereotype.Service;
 
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Set;
 import java.util.stream.Collectors;
```
Comment (Andrii77711) [inline src/main/java/com/to/ordermanagementservice/service/impl/SimpleOrderService.java:14]: Ya pro eto govoru

File: src/main/java/com/to/ordermanagementservice/service/impl/SimpleOrderService.java (hunk 1 above)
Comment (taras-oglabyak) [reply src/main/java/com/to/ordermanagementservice/service/impl/SimpleOrderService.java]: dfdsfsdfsfsdfdfsf

File: src/main/java/com/to/ordermanagementservice/service/impl/SimpleOrderService.java [hunk 5]
```diff
@@ -11,6 +11,7 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Collectors;
```
Comment (taras-oglabyak) [inline src/main/java/com/to/ordermanagementservice/service/impl/SimpleOrderService.java]: is this import used?

File: src/main/java/com/to/ordermanagementservice/service/impl/SimpleOrderService.java [hunk 6]
```diff
@@ -46,20 +47,19 @@ private OrderDetails collectOrderDetails(Order order) {
...
-            orderItemDetails.setUpdateAt(orderItem.getUpdatedAt());
-            orderItemDetails.setCreatedAt(orderItem.getCreatedAt());
-            orderItemDetails.setPrice(orderItem.getPrice());
-            orderItemDetails.setQuantity(orderItem.getQuantity());
-            orderItemDetails.setProductId(orderItem.getProductId());
-            result.add(orderItemDetails);
-        }
-        return result;
+        return orderItemRepository.getOrderItemsByOrderId(orderId).stream()
+                .map(orderItem -> convertOrderItemToOrderItemDetails(orderItem)).toList();
```
Comment (taras-oglabyak) [inline src/main/java/com/to/ordermanagementservice/service/impl/SimpleOrderService.java]: the good idea would be to keep each method call for streams in serapate lines. e.g:
```java
return collection.stream()
    .filter()
    .map()
    .sorted()
    .collect();
```

**Review by taras-oglabyak — CHANGES_REQUESTED** [review]

File: src/main/java/com/to/ordermanagementservice/service/impl/SimpleOrderService.java (hunk 6 above)
Comment (Andrii77711) [reply src/main/java/com/to/ordermanagementservice/service/impl/SimpleOrderService.java]: understand

File: src/main/java/com/to/ordermanagementservice/service/impl/SimpleOrderService.java (hunk 5 above)
Comment (Andrii77711) [reply src/main/java/com/to/ordermanagementservice/service/impl/SimpleOrderService.java]: no and delete

**Review by taras-oglabyak — APPROVED** [review]

**Pull request merged**